src/workspace-data/data-table/wds/RecordDeleter.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { b, div, h } from 'react-hyperscript-helpers';
+import { absoluteSpinnerOverlay, DeleteConfirmationModal } from 'src/components/common';
+import { Ajax } from 'src/libs/ajax';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import * as Utils from 'src/libs/utils';
+
+export const RecordDeleter = ({ onDismiss, onSuccess, dataProvider, collectionId, selectedRecords, runningSubmissionsCount }) => {
+  const [additionalDeletions, setAdditionalDeletions] = useState([]);
+  const [deleting, setDeleting] = useState(false);
+
+  const selectedKeys = _.keys(selectedRecords);
+
+  const doDelete = async () => {
+    const recordsToDelete = _.flow(
+      _.map(({ name: entityName, entityType }) => ({ entityName, entityType })),
+      (records) => _.concat(additionalDeletions, records)
+    )(selectedRecords);
+
+    const recordTypes = _.uniq(_.map(({ entityType }) => entityType, selectedRecords));
+    if (recordTypes.length > 1) {
+      await reportError('Something went wrong; more than one recordType is represented in the selection. This should not happen.');
+    }
+    const recordType = recordTypes[0];
+    setDeleting(true);
+
```

Comment:
Add a new function `filterAdditionalDeletions` to ensure the logic is testable independently. 

```suggestion

const filterAdditionalDeletions = async (error: Response, recordsToDelete: Array<{ entityType: string, entityName: string }>) => {
  const errorEntities = await error.json();
  
  return _.filter(errorEntities, (errorEntity: { entityType: string, entityName: string }) => 
    !_.some(recordsToDelete, (selectedEntity) => 
      selectedEntity.entityType === errorEntity.entityType && selectedEntity.entityName === errorEntity.entityName
    )
  );
};
```